cloud/ec2_fleet.go

```diff
@@ -0,0 +1,538 @@
+			"host_provider": h.Distro.Provider,
+			"distro":        h.Distro.Id,
+		}))
+		return nil, errors.Wrapf(err, "failed to attach tags for %s", h.Id)
+	}
+	grip.Debug(message.Fields{
+		"message":       "attached tags for host",
+		"host":          h.Id,
+		"host_provider": h.Distro.Provider,
+		"distro":        h.Distro.Id,
+	})
+
+	event.LogHostStarted(h.Id)
+	return h, nil
+}
+
+func (m *ec2FleetManager) GetInstanceStatuses(ctx context.Context, hosts []host.Host) ([]CloudStatus, error) {
+	instanceIDs := make([]*string, 0, len(hosts))
+	for _, h := range hosts {
+		instanceIDs = append(instanceIDs, aws.String(h.Id))
+	}
+
+	if err := m.client.Create(m.credentials, defaultRegion); err != nil {
+		return nil, errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	describeInstancesOutput, err := m.client.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+		InstanceIds: instanceIDs,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "error describing instances")
+	}
+
+	if err = validateEc2DescribeInstancesOutput(describeInstancesOutput); err != nil {
+		return nil, errors.Wrap(err, "invalid describe instances response")
+	}
+
+	if len(instanceIDs) != len(describeInstancesOutput.Reservations) {
+		return nil, errors.Errorf("AWS returned %d statuses for %d hosts", len(describeInstancesOutput.Reservations), len(instanceIDs))
+	}
+
+	statusMap := map[string]CloudStatus{}
+	for i := range describeInstancesOutput.Reservations {
+		statusMap[*describeInstancesOutput.Reservations[i].Instances[0].InstanceId] = ec2StatusToEvergreenStatus(*describeInstancesOutput.Reservations[i].Instances[0].State.Name)
+	}
+
+	// Return as an ordered slice of statuses
+	statuses := []CloudStatus{}
+	for _, h := range hosts {
+		statuses = append(statuses, statusMap[h.Id])
+	}
+	return statuses, nil
+}
+
+func (m *ec2FleetManager) GetInstanceStatus(ctx context.Context, h *host.Host) (CloudStatus, error) {
+	r, err := getRegion(h)
+	if err != nil {
+		return StatusUnknown, errors.Wrap(err, "problem getting region from host")
+	}
+	if err = m.client.Create(m.credentials, r); err != nil {
+		return StatusUnknown, errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	instance, err := m.getInstance(ctx, h)
+	if err != nil {
+		return StatusUnknown, errors.Wrapf(err, "can't get instance status")
+	}
+
+	return ec2StatusToEvergreenStatus(*instance.State.Name), nil
+}
+
+func (m *ec2FleetManager) TerminateInstance(ctx context.Context, h *host.Host, user string) error {
+	if h.Status == evergreen.HostTerminated {
+		return errors.Errorf("Can not terminate %s - already marked as terminated!", h.Id)
+	}
+	r, err := getRegion(h)
+	if err != nil {
+		return errors.Wrap(err, "problem getting region from host")
+	}
+	if err = m.client.Create(m.credentials, r); err != nil {
+		return errors.Wrap(err, "error creating client")
+	}
+	defer m.client.Close()
+
+	resp, err := m.client.TerminateInstances(ctx, &ec2.TerminateInstancesInput{
```

Comment:
Does this delete the fleet request as well, as in the `DeleteFleets` API call? I think the fleet request will remain active if not deleted.

Reply:
It doesn't delete the Fleet.
My original plan was to: 1) upload a launch template, 2) create a Fleet, 3) delete the Fleet and 4) delete the template. I ended up skipping #3 since AWS would not let me delete a Fleet in instant  mode:

`Unsupported: Fleet fleet-abcd-efghi is instant fleet. This type of fleet cannot be cancelled`. 

My understanding is that an instant fleet request goes away on its own once the request is fulfilled, which happens synchronously with the API call. (This is another reason I wasn't worried about the 1000 Fleet limit). I don't see the Fleet requests in the AWS console.